server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/SubAddressing.java

```diff
@@ -55,20 +63,36 @@ public void service(Mail mail) throws MessagingException {
                 .ifPresent(
                     Throwing.consumer(details -> {
 
-                        boolean hasRight = mailboxManager.hasRight(
-                            MailboxPath.forUser(
-                                Username.fromMailAddress(recipient.stripDetails("+")),
-                                details
-                            ),
-                            MailboxACL.Right.Post,
-                            session
-                        );
+                        boolean hasRight;
+
+                        try {
+                            Username recipientUsername = Username.fromLocalPartWithoutDomain(recipient.getLocalPartWithoutDetails("+"));
+                            MailboxSession session = mailboxManager.createSystemSession(recipientUsername);
 
-                        String targetFolder = hasRight ? details : "inbox";
+                            MailboxACL aclCommand = mailboxManager.getMailbox(
+                                    MailboxPath.forUser(recipientUsername, details), session)
+                                .getMetaData(IGNORE, session, MessageManager.MailboxMetaData.FetchGroup.NO_COUNT)
+                                .getACL();
 
+                            hasRight = aclCommand.getEntries().containsKey(MailboxACL.ANYBODY_KEY) &&
```

Comment:
Do not re-allocate variables

Generally method extraction is a good alternative.